### PR TITLE
CRO i5: Add new plans filter bar.

### PR DESCRIPTION
**Pull request by elliottprogrammer**
### Changes proposed in this Pull Request

This PR add the new CRO i5 iteration of the plans page filter bar for Calypso and Jetpack Cloud.

Fixes 1196341175636977-as-1199149328812588

#### Implementation notes:

- I did not implement the new scroll behavior is this PR (only pin the plans filter bar. The page heading and navigation should not also be pinned but rather disappear when scrolled)

### Testing instructions

- Download the PR
- Run both Calypso and Jetpack cloud concurrently
- Select the `i5 - Saas table design` variant of the a/b test jetpackConversionRateOptimization
- Visit the Plans page
- In both Calypso and Jetpack Cloud, check:
    - That the plans filter bar matches the Figma design (In desktop & mobile)
    - That the Filter Bar duration toggle functions as it should (Changes the plans/products billing duration and prices)
    - Select the previous a/b variant (v2 - slide outs) and check that the previous plans filter bar iteration was not affected.

### Figma designs

Desktop
Screenshot on 2020-11-11 at 08-44-49

Desktop scrolled
Screenshot on 2020-11-11 at 08-45-55

Mobile
Screenshot on 2020-11-11 at 08-47-14

Mobile scrolled
Screenshot on 2020-11-11 at 08-47-53

.
.
### This PR screenshots:

Desktop
Screenshot on 2020-11-11 at 08-48-27

Desktop scrolled
Screenshot on 2020-11-11 at 08-49-11

Mobile
Screenshot on 2020-11-11 at 08-49-52

Mobile scrolled
Screenshot on 2020-11-11 at 08-50-25

### CODE REVIEW

File: client/my-sites/plans-v2/plans-filter-bar-i5/index.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	if ( ! highestAnnualDiscount ) {
+		return null;
+	}
+
+	return (
+		<div className={ classNames( 'plans-filter-bar-i5__discount-message', { primary: primary } ) }>
+			<div>
+				<span className="message">
+					{ isMobile
+						? translate( 'SAVE %(discount)s BY PAYING YEARLY', {
```
Comment (a8ci18n) [inline client/my-sites/plans-v2/plans-filter-bar-i5/index.tsx:78]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **7** times:
`translate( 'Save %(discount)s by paying yearly' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/plans-filter-bar-i5)

File: client/my-sites/plans-v2/plans-filter-bar-i5/index.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+								args: { discount: highestAnnualDiscount },
+								comment: 'Discount is either a currency-formatted number or percentage',
+						  } )
+						: translate( 'Save %(discount)s', {
+								args: { discount: highestAnnualDiscount },
+								comment: 'Discount is either a currency-formatted number or percentage',
+						  } ) }
+				</span>
+				🎉
```
Comment (elliottprogrammer) [inline client/my-sites/plans-v2/plans-filter-bar-i5/index.tsx:87]: I'm wondering if there is a better way to display this emoji..  Image? Or maybe unicode?  I'll look into this..

File: client/my-sites/plans-v2/types.ts [hunk 3]
```diff
@@ -130,3 +132,15 @@ export interface AvailableProductData {
 	product_slug: string;
 	cost: number;
 }
+
+export interface FilterBarProps {
```
Comment (monsieur-z) [inline client/my-sites/plans-v2/types.ts:136]: For readability, I'd day the component props should stay in the component file unless they're imported elsewhere.

File: client/my-sites/plans-v2/products-grid-i5/style.scss [hunk 4]
```diff
@@ -29,3 +33,8 @@
 		grid-gap: 24px 16px;
 	}
 }
+
+.products-grid-i5__filter-bar {
+	height: 63px;
```
Comment (monsieur-z) [inline client/my-sites/plans-v2/products-grid-i5/style.scss:38]: Out of curiosity, why do we hardcode a height value? Is it necessary when we want the filter bar to stick?

File: client/my-sites/plans-v2/plans-filter-bar-i5/index.tsx (hunk 2 above)
Comment (monsieur-z) [reply client/my-sites/plans-v2/plans-filter-bar-i5/index.tsx:87]: Seems that using emojis is well supported, from what I read :)

File: client/my-sites/plans-v2/plans-filter-bar-i5/index.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+const CALYPSO_MASTERBAR_HEIGHT = 47;
+const CLOUD_MASTERBAR_HEIGHT = 94;
+
+const REALTIME_PRODUCTS = [ PLAN_JETPACK_SECURITY_REALTIME, PRODUCT_JETPACK_BACKUP_REALTIME ];
+
+const DiscountMessage: React.FC< DiscountMessageProps > = ( { primary } ) => {
+	const translate = useTranslate();
+	const isMobile: boolean = useMobileBreakpoint();
+
+	const croVersion = getJetpackCROActiveVersion();
```
Comment (monsieur-z) [inline client/my-sites/plans-v2/plans-filter-bar-i5/index.tsx:48]: That's most likely not necessary, since we're in a i5 specific component.

**Review by monsieur-z — APPROVED** [review]
Nice! Works well :)

I left a couple of non blocker comments. Your implementation is a little bit different than the mockups (in which discount text is uppercase), but from reading the comments in the project thread it seems that it might change. I'd sync with @eeeeevon13.

File: client/my-sites/plans-v2/products-grid-i5/style.scss (hunk 4 above)
Comment (elliottprogrammer) [reply client/my-sites/plans-v2/products-grid-i5/style.scss:38]: This is the wrapper div around the filter-bar, so when the inner filter-bar goes position fixed (and is suddenly removed from the document flow), this wrapper div stays static and maintains the height of the filter-bar, preventing the page from "jumping-up".  Another way to achieve this jumping prevention would be with css, adding margin, or padding, or translateY when the when the 'sticky' class is added. Either way, they are both hard-coded. I'm not sure if one is better than the other. I'm now wondering if there is a better way to achieve this. I'll look into it. If you are aware of a better solution here, please share 🙂

File: client/my-sites/plans-v2/plans-filter-bar-i5/index.tsx (hunk 2 above)
Comment (elliottprogrammer) [reply client/my-sites/plans-v2/plans-filter-bar-i5/index.tsx:87]: Thanks @monsieur-z , yes it appears so, and it looks like there are some other methods that can be used too.  HTML entity `&#xF389;` or JS `String.fromCodePoint(0x1F389)` , But I don't see that one has any advantage over the other in this case. 👍

**Comment by elliottprogrammer** [comment]
> Your implementation is a little bit different than the mockups (in which discount text is uppercase), but from reading the comments in the project thread it seems that it might change.

Oh, nice catch, thank you. I didn't mean to have it lowercase here.  I'll change it to uppercase for now.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/5176836

Thank you @elliottprogrammer for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/5176836).

**Pull request merged**